processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -199,6 +192,15 @@ protected Integer addToFacts(
       if (prev != null) {
         rowOffset = prev;
       } else {
+        rowContainer.set(row);
+        for (int i = 0; i < metrics.length; i++) {
+          final AggregatorFactory agg = metrics[i];
+          aggs[i] = agg.factorizeBuffered(
+              makeColumnSelectorFactory(agg, rowSupplier, deserializeComplexMetrics)
+          );
+        }
+        rowContainer.set(null);
```

Comment:
same rowContainer set/unset happens at line210 and 216 also, can we do the rowContainer.set and unset once only?